internal/statemachine/common_states.go

```diff
@@ -3,29 +3,34 @@ package statemachine
 import (
 	"fmt"
 	"os"
+
+	"github.com/google/uuid"
 )
 
 // generate work directory file structure
 func (stateMachine *StateMachine) makeTemporaryDirectories() error {
 	// if no workdir was specified, open a /tmp dir
 	if stateMachine.stateMachineFlags.WorkDir == "" {
-		workDir, err := os.MkdirTemp(stateMachine.tempLocation, "ubuntu-image-")
-		if err != nil {
-			return fmt.Errorf("Failed to create temporary directory")
+		stateMachine.stateMachineFlags.WorkDir = "ubuntu-image-" + uuid.NewString()
```

Comment:
You are understanding this correctly, and it is a mistake. I've actually already fixed it on the load-gadget-yaml branch, but forgot to push it here. I will do that now.

Just for some more background on why this change was made: os.MkdirTemp always makes directories in "/tmp". I was using that for its simplicity and convenience, but it was only added to golang in version 1.16. I changed to using os.Mkdir (and a few other minor changes) to ensure it could build on focal, which has go version 1.13.